### PR TITLE
[SPARK-45202][BUILD] Fix lint-js tool and js format

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

This PR fixes lint-js tool, which covers only js files in the core module, and also fixes the js format of the rest modules.

```
dev/lint-js

/Users/kentyao/spark/sql/core/src/main/resources/org/apache/spark/sql/execution/ui/static/spark-sql-viz.js
   29:10  error  'renderPlanViz' is defined but never used                                      no-unused-vars
   31:18  error  'd3' is not defined                                                            no-undef
   35:11  error  'graphlibDot' is not defined                                                   no-undef
   37:22  error  'dagreD3' is not defined                                                       no-undef
   46:27  error  '$' is not defined                                                             no-undef
   59:38  error  'd3' is not defined                                                            no-undef
   66:21  error  'd3' is not defined                                                            no-undef
   67:3   error  'd3' is not defined                                                            no-undef
   68:20  error  'd' is defined but never used. Allowed unused args must match /^_ignored_.*/u  no-unused-vars
   69:21  error  'd3' is not defined                                                            no-undef
   70:7   error  '$' is not defined                                                             no-undef
  104:55  error  'i' is defined but never used. Allowed unused args must match /^_ignored_.*/u  no-unused-vars
  108:1   error  Expected indentation of 10 spaces but found 12                                 indent
  109:1   error  Expected indentation of 10 spaces but found 12                                 indent
  115:1   error  Expected indentation of 4 spaces but found 6                                   indent
  116:1   error  Expected indentation of 6 spaces but found 8                                   indent
  116:16  error  'd3' is not defined                                                            no-undef
  117:1   error  Expected indentation of 4 spaces but found 6                                   indent
  118:1   error  Expected indentation of 2 spaces but found 4                                   indent
  129:13  error  'd3' is not defined                                                            no-undef
  130:34  error  'd3' is not defined                                                            no-undef
  133:13  error  'd3' is not defined                                                            no-undef
  134:34  error  'd3' is not defined                                                            no-undef
  137:13  error  'd3' is not defined                                                            no-undef
  138:15  error  'd3' is not defined                                                            no-undef
  142:13  error  'd3' is not defined                                                            no-undef
  143:15  error  'd3' is not defined                                                            no-undef
  180:11  error  'd3' is not defined                                                            no-undef
  196:3   error  '$' is not defined                                                             no-undef
  198:26  error  '$' is not defined                                                             no-undef
  199:1   error  Expected indentation of 6 spaces but found 8                                   indent
  200:1   error  Expected indentation of 8 spaces but found 10                                  indent
  201:1   error  Expected indentation of 8 spaces but found 10                                  indent
  201:28  error  'd3' is not defined                                                            no-undef
  201:41  error  '$' is not defined                                                             no-undef
  202:1   error  Expected indentation of 8 spaces but found 10                                  indent
  203:1   error  Expected indentation of 8 spaces but found 10                                  indent
  204:1   error  Expected indentation of 8 spaces but found 10                                  indent
  204:11  error  '$' is not defined                                                             no-undef
  205:1   error  Expected indentation of 6 spaces but found 8                                   indent
  206:1   error  Expected indentation of 8 spaces but found 10                                  indent
  207:1   error  Expected indentation of 6 spaces but found 8                                   indent
  208:1   error  Expected indentation of 4 spaces but found 2                                   indent
  210:22  error  '$' is not defined                                                             no-undef
  212:1   error  Expected indentation of 4 spaces but found 6                                   indent
  212:40  error  '$' is not defined                                                             no-undef
  223:27  error  '$' is not defined                                                             no-undef

✖ 47 problems (47 errors, 0 warnings)
  17 errors and 0 warnings potentially fixable with the `--fix` option.


/Users/kentyao/spark/docs/js/api-docs.js
  20:1   error  '$' is not defined                                                                 no-undef
  21:21  error  '$' is not defined                                                                 no-undef
  29:14  error  '$' is not defined                                                                 no-undef
  32:22  error  'index' is defined but never used. Allowed unused args must match /^_ignored_.*/u  no-unused-vars
  33:19  error  '$' is not defined                                                                 no-undef
  35:5   error  '$' is not defined                                                                 no-undef
  45:1   error  '$' is not defined                                                                 no-undef
  50:5   error  'MathJax' is not defined                                                           no-undef

/Users/kentyao/spark/docs/js/api-javadocs.js
  20:1   error  '$' is not defined                                                                 no-undef
  27:14  error  '$' is not defined                                                                 no-undef
  30:22  error  'index' is defined but never used. Allowed unused args must match /^_ignored_.*/u  no-unused-vars
  31:19  error  '$' is not defined                                                                 no-undef
  33:5   error  '$' is not defined                                                                 no-undef
  37:22  error  'index' is defined but never used. Allowed unused args must match /^_ignored_.*/u  no-unused-vars
  38:9   error  '$' is not defined                                                                 no-undef
  39:7   error  '$' is not defined                                                                 no-undef
  40:16  error  '$' is not defined                                                                 no-undef
  44:30  error  '$' is not defined                                                                 no-undef
  54:16  error  '$' is not defined                                                                 no-undef
  55:7   error  '$' is not defined                                                                 no-undef
  57:7   error  '$' is not defined                                                                 no-undef

/Users/kentyao/spark/docs/js/main.js
   28:3   error  '$' is not defined                no-undef
   29:5   error  '$' is not defined                no-undef
   32:18  error  '$' is not defined                no-undef
   33:5   error  '$' is not defined                no-undef
   36:23  error  '$' is not defined                no-undef
   40:7   error  '$' is not defined                no-undef
   41:7   error  '$' is not defined                no-undef
   43:18  error  '$' is not defined                no-undef
   44:19  error  '$' is not defined                no-undef
   45:20  error  '$' is not defined                no-undef
   50:7   error  '$' is not defined                no-undef
   54:13  error  'buttonLabel' is already defined  no-redeclare
   56:13  error  'buttonLabel' is already defined  no-redeclare
   63:9   error  '$' is not defined                no-undef
   75:3   error  '$' is not defined                no-undef
   79:24  error  '$' is not defined                no-undef
   79:47  error  '$' is not defined                no-undef
   80:5   error  '$' is not defined                no-undef
   80:13  error  '$' is not defined                no-undef
   81:5   error  '$' is not defined                no-undef
   81:27  error  '$' is not defined                no-undef
   89:31  error  '$' is not defined                no-undef
   90:18  error  '$' is not defined                no-undef
   91:5   error  '$' is not defined                no-undef
   95:1   error  '$' is not defined                no-undef
   99:3   error  'anchors' is not defined          no-undef
  102:3   error  'anchors' is not defined          no-undef
  104:3   error  '$' is not defined                no-undef
  110:3   error  '$' is not defined                no-undef
  115:21  error  '$' is not defined                no-undef
  116:27  error  '$' is not defined                no-undef
  117:25  error  '$' is not defined                no-undef
  119:3   error  '$' is not defined                no-undef
  123:25  error  '$' is not defined                no-undef
  129:25  error  '$' is not defined                no-undef

✖ 56 problems (56 errors, 0 warnings)
```

### Why are the changes needed?

bugfix and format fix

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

dev/lint-js in CI

### Was this patch authored or co-authored using generative AI tooling?

no

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @sarutak @dongjoon-hyun @HyukjinKwon, thanks

File: dev/lint-js [hunk 1]
```diff
@@ -44,8 +44,14 @@ if ! npm ls eslint > /dev/null; then
   npm ci eslint
 fi
 
-npx eslint -c "$SPARK_ROOT_DIR/dev/eslint.json" $LINT_TARGET_FILES | tee "$LINT_JS_REPORT_FILE_NAME"
```
[the remark this answers is not in the document]
Comment (dongjoon-hyun) [reply dev/lint-js:47]: +1 for the above comment.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (except the above comments about `lint-js`)

**Review by HyukjinKwon — APPROVED** [review]

File: dev/lint-js [hunk 2]
```diff
@@ -44,8 +44,14 @@ if ! npm ls eslint > /dev/null; then
...
-lint_status=$?
+declare lint_status=0
+for dir in "${LINT_TARGET_FILES[@]}"; do
+  npx eslint -c "$SPARK_ROOT_DIR/dev/eslint.json" "$dir"
+  status=$?
+  if [ $status -ne 0 ] ; then
+    lint_status=$status
+  fi
+done
```
Comment (yaooqinn) [inline dev/lint-js:54]:
```suggestion
npx eslint -c "$SPARK_ROOT_DIR/dev/eslint.json" ${LINT_TARGET_FILES[@]} | tee "$LINT_JS_REPORT_FILE_NAME"
lint_status=$?
```

File: dev/lint-js (hunk 1 above)
[the remark this answers is not in the document]
Comment (yaooqinn) [reply dev/lint-js:47]: SGTM, Thanks for the suggestion

**Comment by yaooqinn** [comment]
The job containing lint-js passed.

Thanks @sarutak @dongjoon-hyun @HyukjinKwon , merged to master

**Pull request closed without merging**